python/plugins/grassprovider/Grass7Utils.py

```diff
@@ -254,8 +254,23 @@ def grassPath():
         return folder or ''
 
     @staticmethod
-    def grassDescriptionPath():
-        return os.path.join(os.path.dirname(__file__), 'description')
+    def userDescriptionFolder():
+        """
+        Creates and returns a directory for users to create additional algorithm descriptions.
+        Or modified versions of stock algorithm descriptions shipped with QGIS.
+        """
+        folder = os.path.join(userFolder(), 'grassaddons', 'description')
+        mkdir(folder)
+        return os.path.abspath(folder)
+
+    @staticmethod
+    def grassDescriptionFolders():
+        """
+        Returns the directories to search for algorithm descriptions.
+        Note that the provider will load from these in sequence, so we put the userDescriptionFolder first
+        to allow users to create modified versions of stock algorithms shipped with QGIS.
+        """
+        return [Grass7Utils.userDescriptionFolder(), os.path.join(os.path.dirname(__file__), 'description')]
```

Comment:
Shall we use `Path` lib instead of `os.path` here and before?

Reply:
OK, I've updated to use Path consistently, although I'm not aware of any benefit to this change.